src/main/java/io/github/syst3ms/skriptparser/expressions/ExprDifference.java

```diff
@@ -62,14 +64,12 @@ public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContex
         return commonSuperClass;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public Object[] getValues(TriggerContext ctx) {
-        Object f = first.getSingle(ctx);
-        Object s = second.getSingle(ctx);
-        if (f == null || s == null)
-            return new Object[0];
-
-        return new Object[] {math.difference(f, s)};
+        return DoubleOptional.of(first.getSingle(ctx), second.getSingle(ctx))
```

Comment:
It needs to be ofOptional(), right now you're using a DoubleOptional<Optional<Object>, Optional<Object>>.